modules/codegen/src/main/scala/com/twilio/guardrail/generators/Http4sServerGenerator.scala

```diff
@@ -77,10 +77,13 @@ object Http4sServerGenerator {
       case GetExtraRouteParams(tracing) =>
         for {
           _ <- Target.log.debug("Http4sServerGenerator", "server")(s"getExtraRouteParams(${tracing})")
-          res <- if (tracing) {
-            Target.pure(List(param"""trace: String => Request[F] => TraceBuilder[F]"""))
-          } else Target.pure(List.empty)
-        } yield res
+          handlerWrapper = param"""handlerWrapper: (String, Request[F], F[Response[F]]) => F[Response[F]] = (_: String, _: Request[F], r: F[Response[F]]) => r"""
```

Comment:
`handlerWrapper` reads as though it's wrapping the entire `Handler`, where as we're just wrapping an individual route or operation. Maybe `mapRoute`?

It's also somewhat undesirable that the user can now arbitrarily transform the response, potentially violating the specification.

Reply:
Additionally, do we need to `(String, Request[F], F[Response[F]]) => F[Response[F]]`, or can we just `(String, Request[F], F[Response[F]]) => F[Unit]` and just `<*` them?

Do you have an example of how this will be used?

Once we're done setting this up here, I'll port the interface we settle on here into the akka-http interface, and @kelnos may be interested in it for Java, though I seem to remember that Dropwizard is more based on registering mutable transformers on the routing core.

Reply:
i like the `mapRoute` name, lets use that

well one of the use cases we thought about was for example when error occurs, we might be able to provide some better error messages in the body (include operation method etc) but we don't have a stronger argument for it. If you feel F[Unit] is better, we can live with it

Reply:
Pls let me know if you'd prefer the `F[Unit]` so that i can change the PR if needed. I think returning `F[Response[F]]` gives more flexibility but i don't have strong feelings either way

Reply:
If we decide to go with `F[Unit]`, we can rename the method to something like `observeRoute`

Reply:
Let's go with `F[Unit]` for now, until we definitely know that we need the more flexibility. `F[Unit]` => `F[Response[F]]` is a fairly straightforward change, adding functionality. It would be much more difficult to go from `F[Response[F]]` to `F[Unit]` once folks start depending on that behaviour.